CorinneTest/Factory.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CorinneTest
+{
+    class Factory : IPersonFactory
+    {
+       
+        static Random rnd = new Random();
+
+
+        public static List<Person> CreateInRange(int lowAge,
+                                                 int highAge,
+                                                 int? numOfPeople)
+        {
+            var list = new List<Person>();
+            int i = 0;
+            while (i < numOfPeople)
+            {
+                int type = rnd.Next(1, 9);
```

Comment:
Use a switch statement?

Reply:
Also, logically, I'm not sure this would compile.  "newPerson" is defined within the if() or else if().  HOw would "list.Add" know about the "newPerson" obj?

Reply:
changed to use switch statement and defined newPerson outside of the switch statement